R/stat-smooth.r

```diff
@@ -127,12 +125,25 @@ StatSmooth <- ggproto("StatSmooth", Stat,
         xseq <- seq(range[1], range[2], length.out = n)
       }
     }
-    # Special case span because it's the most commonly used model argument
+
+    ## Special case span because it's the most commonly used model argument
     if (identical(method, "loess")) {
       method.args$span <- span
     }
 
-    if (is.character(method)) method <- match.fun(method)
+    ## If gam and gam's method is not specified by the user then use REML
+    if (identical(method, "gam") && is.null(method.args$method)) {
```

Comment:
This check should probably be `identical(method, mgcv::gam)` and happen after the conversion from character to function in the next block.

Reply:
I changed the design slightly here. In master `params$method` was replaced by `mgcv::gam` if it was equal to `"gam"`, withiin `setup_params`. Now `params$method` keeps being `gam` and matching names to functions happens in `compute_group`. This is to preserve symmetry with how the other methods are handled (e.g. `loess`) where `method` is `match.fun`ed. 

The `is.character` check may as well be redundant but I kept it in just in case I was breaking anything I could not foresee. 

Happy to revert to the earlier design and test functions being identical instead of character strings, if this causes any issues; though `ggplot2` tests were successful my side.

Reply:
Or perhaps an alternative closer to what you are suggesting, that avoids checking twice is method is `"gam"` can be something like:

```r
    if (is.character(method)) {
        if (identical(method, "gam")) {
            method <- mgcv::gam
            if (is.null(method.args$method)) {
                method.args$method <- "REML"
            }
        } else {
            method <- match.fun(method)
        }
    }
```

Again the `is.character` may not be needed anymore.

Reply:
OK; just figured out that ggplot2 wants the user to be able to supply functions for `method` as well as character strings so checking `method` against `mgcv::gam` after matching makes more sense. Just pushed a fix.